### DISCUSSION TITLE
Block video upload (or even selection) on Mobile app

### DISCUSSION THREAD

**Discussion by neopit** [GetStream/stream-chat-react-native#1711] [Q&A] [answered]
Is there a way to block video upload (or event earlier video selection) on Mobile?

**Comment by vishalnarkhede** [comment, marked as answer]
@neopit To disallow video upload, please take a look at [How to block certain file extensions from uploads](https://getstream.io/chat/docs/javascript/app_setting_overview/?language=javascript#file-uploads).

But to disallow video selection in media picker, you will need to [override native handler](https://getstream.io/chat/docs/sdk/reactnative/customization/native_handlers/#overriding-handlers), specifically [getPhotos](https://getstream.io/chat/docs/sdk/reactnative/customization/native_handlers/#getphotos) handler.

> Following implementation for getPhotos handler has been pretty much copied from [default handler](https://github.com/GetStream/stream-chat-react-native/blob/develop/package/native-package/src/index.js#L56) except for the line which has a comment next to it.

```tsx
import { registerNativeHandlers } from 'stream-chat-react-native';
import { Image, PermissionsAndroid, Platform } from 'react-native';
import { CameraRoll } from '@react-native-camera-roll/camera-roll';

registerNativeHandlers({
  getPhotos: async ({ after, first }) => {
    try {
      if (Platform.OS === 'android') {
        const readExternalStoragePermissionAndroid =
          PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE;
        const hasPermission = await PermissionsAndroid.check(readExternalStoragePermissionAndroid);
        if (!hasPermission) {
          const granted = await PermissionsAndroid.request(readExternalStoragePermissionAndroid, {
            buttonNegative: 'Deny',
            buttonNeutral: 'Ask Me Later',
            buttonPositive: 'Allow',
            message: 'Permissions are required to access and share photos.',
            title: 'Photos Access',
          });
          if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
            throw new Error('getPhotos Error');
          }
        }
      }
      const results = await CameraRoll.getPhotos({
        after,
        assetType: 'Photos', // <-- THIS IS THE ONLY CHANGE HERE.
        first,
        include: ['fileSize', 'filename', 'imageSize', 'playableDuration'],
      });
      const assets = results.edges.map((edge) => ({
        ...edge.node.image,
        source: 'picker',
      }));
      const hasNextPage = results.page_info.has_next_page;
      const endCursor = results.page_info.end_cursor;
      return { assets, endCursor, hasNextPage };
    } catch (_error) {
      throw new Error('getPhotos Error');
    }
  },
});
```

**Reply by neopit** [reply]
Thx for the solution

However, using registerNativeHandlers seems to affect the entire app not only the picker at the Chat level if I am not wrong

**Reply by vishalnarkhede** [reply]
No, `registerNativeHandlers` is only used in chat related code. It shouldn't affect any other functionality of the application.

**Reply by neopit** [reply]
Thx for the info
